public_html/admin/controller/pages/sale/contact.php

```diff
@@ -109,30 +100,30 @@ public function main()
             $this->data['success'] = '';
         }
 
-        $this->data['action'] = $this->html->getSecureURL('sale/contact');
-        $this->data['cancel'] = $this->html->getSecureURL('sale/contact');
+        $this->data['action'] = $this->data['cancel'] = $this->html->getSecureURL('sale/contact');
 
         //get store from main switcher and current config
-        $this->data['store_id'] = (int)$this->session->data['current_store_id'];
+        $this->data['store_id'] = $this->session->data['current_store_id'] ?? 0;
+        $this->data['customers'] = $this->data['products'] = [];
 
-        $this->data['customers'] = array();
-        $this->data['products'] = array();
         $this->loadModel('catalog/product');
         $customer_ids = $this->request->get_or_post('to');
-        if (!$customer_ids && has_value($this->session->data['sale_contact_presave']['to'])) {
-            $customer_ids = $this->session->data['sale_contact_presave']['to'];
+        if (!$customer_ids && isset($this->session->data['sale_contact_presave']['to'])) {
```

Comment:
what is a result for isset(0) or isset(' ')  ?

Reply:
will be true in both cases. isset(NULL) will be false

Reply:
Hmm. $var = null; is set in reality.

Reply:
nope. https://www.php.net/manual/en/function.isset.php